lib/led-matrix/include/LEDMatrix.h

```diff
@@ -0,0 +1,43 @@
+/* LED Matrix Header */
+#pragma once
+#include "hw_bridge.h"
+class LEDMatrix {
+ private:
+  PwmOut m_RChannel;
+  PwmOut m_GChannel;  
+  PwmOut m_BChannel;
+
+  HWBRIDGE::LEDMATRIX::color flashing_color;
+  uint8_t flashing_red, flashing_green, flashing_blue;
+
+  //float flash_period_sec;
+  Thread *lightsThread;
+  Semaphore reqEndFlash;
+  static constexpr auto period_delay = 500ms; //std chrono literal
+  void terminateFlashing();
+  void flashing();
+
+ public:
+  // Constructor: An LED Matrix is defined by the pins it is connected to.
+  LEDMatrix(PinName R, PinName G, PinName B);
```

Comment:
u should add a destructor that calls terminate flashing if lightsthread != nullptr